Add `Schema` type (#321)
sindresorhus/type-fest: readme.md

- [`SetReturnType`](source/set-return-type.d.ts) - Create a function type with a return type of your choice and the same parameters as the given function type.
- [`Simplify`](source/simplify.d.ts) - Useful to flatten the type output to improve type hints shown in editors. And also to transform an interface into a type to aide with assignability.
- [`Get`](source/get.d.ts) - Get a deeply-nested property from an object using a key path, like [Lodash's `.get()`](https://lodash.com/docs/latest#get) function.

> - [`Schema`](source/schema.d.ts) - Recursively replaces all the property values into a given value type for an object.

Reviewer: The description makes it sound like it mutates the existing type, but it creates a new type. See the description of the other "deep" types.

- [`Includes`](source/includes.d.ts) - Returns a boolean for whether the given array includes the given item.
- [`Simplify`](source/simplify.d.ts) - Useful to flatten the type output to improve type hints shown in editors. And also to transform an interface into a type to aide with assignability.
- [`Jsonify`](source/jsonify.d.ts) - Transform a type to one that is assignable to the `JsonValue` type.

> - [`Schema`](source/schema.d.ts) - Deeply changes the values for all properties on a type to whatever specified. Useful for code e.g validation or configuration that relies on using the same structure of a type but the values are different.

Reviewer: ```suggestion
- [`Schema`](source/schema.d.ts) - Deeply change the type of values for all properties on a type to the given value.
```

Reviewer: I think it could still be made clearer though. If you have any suggestions.

Author: @sindresorhus Made changes to it. What do you think about it now?
